v0/destinations/kafka/transform.js

```diff
@@ -1,36 +1,83 @@
+/* eslint-disable no-restricted-syntax */
+const groupBy = require("lodash/groupBy");
+const cloneDeep = require("lodash/cloneDeep");
+const { getDynamicMeta } = require("../../../adapters/utils/networkUtils");
 const { getIntegrationsObj } = require("../../util");
+const { TRANSFORMER_METRIC } = require("../../util/constant");
+const ErrorBuilder = require("../../util/error");
 
 function batch(destEvents) {
   const respList = [];
-  const batchedRequest = [];
-  const metadata = [];
-  destEvents.forEach(event => {
-    metadata.push(event.metadata);
-    batchedRequest.push(event.message);
-  });
-  respList.push({
-    batchedRequest,
-    metadata,
-    destination: destEvents[0].destination
+
+  // Grouping the events by topic
+  const groupedEvents = groupBy(destEvents, event => {
+    return event.message.topic;
```

Comment:
Here the topic is taken from the message where it is picked from the integrations object in the `process` function. Is that a mistake? If no, what's the reason?

Reply:
This is batch api @saikumarrs so by then it is already processed so topic is now available in message.